### PR TITLE
Add support for setting a channel binding token

### PR DESCRIPTION
**Pull request by inejge**
Background: I'm adding GSSAPI support to my [LDAP library](https://github.com/inejge/ldap3), and I want to support the use of channel bindings in the authentication exchange, for interoperating with Active Directory servers which can force channel binding use on TLS LDAP connections as a mitigation against authentication relaying attacks. (AD uses the hash of the server's certificate to create and expect a __tls-server-end-point__ token.)

This patch adds a method (`set_cb_token()`) to set a channel binding token on a `ClientCtx`, which will be consumed with the next `step()` call on the context. It doesn't change the operation of tokenless context initialization in any way, so, if accepted, it should be semver-compatible with the current 0.5 series.

My use case needs a separate (and a bit more intrusive) patch to `cross-krb5`, which I will post shortly.

### CODE REVIEW

File: libgssapi/src/context.rs [hunk 1]
```diff
@@ -697,11 +699,35 @@ impl ClientCtx {
     /// number of times until step returns `Ok(None)`. At that point
     /// the context is fully initialized.
     pub fn step(&mut self, tok: Option<&[u8]>) -> Result<Option<Buf>, Error> {
```
Comment (estokes) [inline libgssapi/src/context.rs]: I think we should just cut 0.6. I'm more concerned with getting the right api in the long run than minimizing churn now. Seems like we should just add an optional 'channel_bindings' arg to step.

File: libgssapi/src/context.rs [hunk 2]
```diff
@@ -747,6 +774,17 @@ impl ClientCtx {
...
     }
+
+    /// Set the channel binding token which will be incorporated
+    /// into the next generated GSSAPI token. Channel bindings are opaque
+    /// structures whose content is dictated by the application.
+    ///
+    /// The channel binding token is "spent" after the `step()` call
+    /// on the context. If you need it for another step, you must
+    /// set it again.
+    pub fn set_cb_token(&mut self, cbt: Vec<u8>) {
```
Comment (estokes) [inline libgssapi/src/context.rs]: Don't force this to be a vec. When you move it to an argument of step make it a Option<&[u8]>

**Review by estokes — CHANGES_REQUESTED** [review]

File: libgssapi/src/context.rs (hunk 1 above)
Comment (inejge) [reply libgssapi/src/context.rs]: :+1:  Patch incoming with the requested changes.

File: libgssapi/src/context.rs (hunk 2 above)
Comment (estokes) [reply libgssapi/src/context.rs]: Looks good

**Pull request merged**